File: 5-longest-palindromic-substring/5-longest-palindromic-substring.java
class Solution {
    int left = 0, right = 0;
    int maxLength = 0;
    public String longestPalindrome(String s) {
        for(int i=0; i<s.length(); i++){
            findPalindrome(s, i, i);
            findPalindrome(s, i, i+1);
        }
        return s.substring(left, right+1);
    }
    private void findPalindrome(String s, int start, int end){
        while(start >=0 && end<s.length() && s.charAt(start) == s.charAt(end)){
            if((end-start+1) > maxLength){
                left = start;
                right = end;
                maxLength = end - start + 1;
            }
            start --;
            end ++;
        }
    }
}

File: 5-longest-palindromic-substring/5-longest-palindromic-substring.py
class Solution:
    left = 0; right = 0
    maxLength = 0
    def find(self, start, end, s):
        while(start >= 0 and end < len(s) and s[start] == s[end]):
            if( (end - start + 1) > self.maxLength ):
                self.left = start
                self.right = end
                self.maxLength = end - start + 1

            start -= 1
            end += 1
    
    def longestPalindrome(self, s: str) -> str:
        for i in range(len(s)):
            self.find(i, i, s)
            self.find(i, i+1, s)
        
        return s[self.left:self.right+1]
    
   
        
        